src/codegen/compile.ml

```diff
@@ -10379,6 +10397,28 @@ and compile_exp_with_hint (env : E.t) ae sr_hint exp =
     )
     ^^
    G.i Unreachable
+
+  | SwitchE (e, cs) when single_case e cs ->
+    let code1 = compile_exp_vanilla env ae e in
+    let [@warning "-8"] [{it={pat={it=TagP (_, pat');_} as pat; exp}; _}] = cs in
```

Comment:
```suggestion
  | SwitchE (e,  [{it={pat={it=TagP (l1, pat'); note = T.Variant [(l1, _)]}; exp }; _} ]  when l1 = l2 ->
    let code1 = compile_exp_vanilla env ae e in
```

Reply:
Dunno. I trust `e`'s type, but not `pat`'s.

Reply:
We do check them, so I think you can trust them. Indeed, it was the checker that caught the bad annoations introduced by pattern inference. But ok, I agree the pattern match is a bit inscrutable.